### ISSUE TITLE
WCAG 2.4.7 AA: Cards have no focus indicator (Issue 30)

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#363]
### Summary

Via Vision Australia assessment: August 2024

Impact: medium

Note: DTA have a 90 day remediation period to address the identified issues within the audit, all issues must be resolved to obtain WCAG 2.2 certification for digital.gov.au.

### Steps to reproduce

[digital.gov.au beta | digital.gov.au](https://www.digital.gov.au/) (https://www.digital.gov.au)

### Observed outcome

The promo cards do not have a visible focus indicator when they receive keyboard focus. This appears to be caused by the overflow: clipped; property on the parent `<div>` element for the links.

![Picture15](https://github.com/user-attachments/assets/255c5b77-80c9-46b8-89da-415b398083fb)

### Code used

```
<div class="ct-promo-card ct-theme-light ct-promo-card--with-image ">
<a href="/initiatives/datadigital">
<div class="ct-promo-card__image">
<img class="ct-image ct-theme-light " src="/sites/default/files/images/2024-06/DTA-home-tile-22.png" alt="&quot;&quot;" aria-hidden="true">
</div>
<div class="ct-promo-card__content">
<h4 class="ct-heading ct-theme-light ct-promo-card__title">
Data and Digital Government Strategy
</h4>
<div class="ct-paragraph ct-theme-light ct-paragraph--regular ct-promo-card__summary">
Guiding government's data and digital transformation initiatives.
</div>
<div class="ct-promo-card__tags">
<span class="ct-link ct-theme-light ct-link--only-icon ct-promo-card__tags__link">
<svg class="ct-icon ct-link__icon" aria-hidden="true" role="img" width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
</svg>
</span>
</div>
</div>
</a>
</div>
CSS
.ct-promo-card {
box-shadow: 0 .25rem 1rem rgba(0,0,0,.2);
position: relative;
border-radius: .25rem;
width: 100%;
overflow: hidden;
}
```

### Why this matters

Without a highly visible focus indicator, sighted keyboard users may have difficulty determining what element has keyboard focus. They may have trouble activating controls or interacting with the content. 

### Expected outcome

Ensure that all focusable elements have a highly visible focus indicator when they receive keyboard focus. Preferably, this should be a solid border that surrounds the component. 

Ensure that the contrast ratio between the focus indicator and the adjacent background is at least 3:1. 

### Code example

```
:focus-visible {
outline: 3px solid black;
box-shadow: 0 0 0 6px white;
outline-offset: .2rem;
}
```
### Additional example

[Service design and delivery process | digital.gov.au](https://www.digital.gov.au/policy/digital-experience/toolkit/service-design-and-delivery-process) (https://www.digital.gov.au/policy/digital-experience/toolkit/service-design-and-delivery-process)

**Description**

The promo cards on the "Service design and delivery process" page do not have a visible focus indicator when they receive keyboard focus. This also appears to be caused by the overflow: clipped; property on the parent `<div>` elements for the links.

![Picture16](https://github.com/user-attachments/assets/04d3d890-f7e6-4b49-b57c-1bca928e89c1)

**Recommendation**

Please refer to the recommendation in the primary example.

**Comment by alan-cole** [comment]
This looks like a customization to promo cards:

![Screenshot 2024-10-14 at 12 03 13 pm](https://github.com/user-attachments/assets/dbbab05b-9050-41b2-af92-c792c52e202c)

Promo cards don't wrap all their content in a link by default; the link is added only to the title (which would display the focus indicator as per other links):

https://github.com/civictheme/uikit/blob/main/components/02-molecules/promo-card/promo-card.twig#L109

**Comment by fionamorrison23** [comment]
Closing this as per Alan Cole's previous remark. This is project-specific, not CivicTheme core.
